frontend/sozisel-app/src/apolloClient.ts

```diff
@@ -1,39 +1,61 @@
-import {AbsintheSocketLink, createAbsintheSocketLink} from "@absinthe/socket-apollo-link";
-import { ApolloClient, ApolloLink, InMemoryCache, NormalizedCacheObject, createHttpLink, split } from '@apollo/client';
+import {
+  AbsintheSocketLink,
+  createAbsintheSocketLink,
+} from "@absinthe/socket-apollo-link";
+import {
+  ApolloClient,
+  ApolloLink,
+  InMemoryCache,
+  NormalizedCacheObject,
+  createHttpLink,
+  split,
+} from "@apollo/client";
 
-import {Socket as PhoenixSocket} from "phoenix";
-import {create as createAbsintheSocket} from "@absinthe/socket";
+import { Socket as PhoenixSocket } from "phoenix";
+import { create as createAbsintheSocket } from "@absinthe/socket";
 import { getMainDefinition } from "@apollo/client/utilities";
+import { setContext } from '@apollo/client/link/context';
 
 const HOST = window.location.hostname;
 
 function createApolloHttpLink(): ApolloLink {
-  return createHttpLink({uri: `http://${HOST}:4000/api`});
+  return createHttpLink({ uri: `http://${HOST}:4000/api` });
 }
 
 function createApolloSocketLink(): ApolloLink {
-  const socket = createAbsintheSocket(new PhoenixSocket(`ws://${HOST}:4000/socket`));
-  
+  const socket = createAbsintheSocket(
+    new PhoenixSocket(`ws://${HOST}:4000/socket`)
+  );
+
   type MockType = AbsintheSocketLink & ApolloLink;
   return createAbsintheSocketLink(socket) as MockType;
 }
-  
 
 export function createApolloClient(): ApolloClient<NormalizedCacheObject> {
-    const splitLink = split(
-      ({ query }) => {
-        const definition = getMainDefinition(query);
-        return (
-          definition.kind === 'OperationDefinition' &&
-          definition.operation === 'subscription'
-        );
-      },
-      createApolloSocketLink(),
-      createApolloHttpLink(),
-    );
-  
-    return new ApolloClient({
-      link: splitLink,
-      cache: new InMemoryCache()
-    });
+  const splitLink = split(
+    ({ query }) => {
+      const definition = getMainDefinition(query);
+      return (
+        definition.kind === "OperationDefinition" &&
+        definition.operation === "subscription"
+      );
+    },
+    createApolloSocketLink(),
+    createApolloHttpLink()
+  );
+
+  const authLink = setContext((_, { headers }) => {
+    const token = localStorage.getItem('token');
+    return {
+      headers: {
+        ...headers,
+        Authorization: token ? `Bearer ${token}` : "",
```

Comment:
We should not set authorization header at all when it is nor present. And moreover this authlink works just for http link and not for websocket link so putting in before websocket link does not change anything (websocket link needs its own token passing when creating connection).